### PR TITLE
Refactor to kotlin.assert and power-assert: example

### PR DESCRIPTION
**Pull request by tibtof**
A first example for #173

### CODE REVIEW

File: src/test/kotlin/io/github/nomisrev/routes/TagRouteSpec.kt [hunk 1]
```diff
@@ -34,8 +30,8 @@ class TagRouteSpec :
       withServer {
         val response = get(TagsResource()) { contentType(ContentType.Application.Json) }
 
-        response.status shouldBe HttpStatusCode.OK
-        response.body<TagsResponse>().tags.toSet() shouldBe emptySet()
+        assert(response.status == HttpStatusCode.OK)
+        assert(response.body<TagsResponse>().tags.isEmpty())
```
Comment (nomisRev) [inline src/test/kotlin/io/github/nomisrev/routes/TagRouteSpec.kt:34]: I think here we need to use `== emptyList()` in order to get a proper error message from power-assert.

```suggestion
        assert(response.body<TagsResponse>().tags == emptyList())
```

File: src/test/kotlin/io/github/nomisrev/routes/TagRouteSpec.kt [hunk 2]
```diff
@@ -10,13 +10,9 @@ import io.github.nomisrev.withServer
...
-import io.kotest.matchers.collections.shouldHaveSize
-import io.kotest.matchers.shouldBe
-import io.ktor.client.call.body
-import io.ktor.client.plugins.resources.get
-import io.ktor.http.ContentType
-import io.ktor.http.HttpStatusCode
-import io.ktor.http.contentType
+import io.ktor.client.call.*
+import io.ktor.client.plugins.resources.*
+import io.ktor.http.*
```
Comment (nomisRev) [inline src/test/kotlin/io/github/nomisrev/routes/TagRouteSpec.kt:15]: I don't use _star_ imports in my projects. Maybe I should add Detekt to check for these cases 🤔

**Review by nomisRev — COMMENTED** [review]
Looks great @tibtof! Two smalls nits.

**Comment by tibtof** [comment]
@nomisRev don't worry about it, and I hope you are well. I will also refactor the other asserts before resubmitting the review.

File: src/test/kotlin/io/github/nomisrev/routes/TagRouteSpec.kt (hunk 1 above)
Comment (tibtof) [reply src/test/kotlin/io/github/nomisrev/routes/TagRouteSpec.kt:34]: I think the message for `isEmpty` is good enough, but if you prefer an equality check, I will refactor it. Here is a comparison:
image

image

File: src/test/kotlin/io/github/nomisrev/routes/TagRouteSpec.kt (hunk 2 above)
Comment (tibtof) [reply src/test/kotlin/io/github/nomisrev/routes/TagRouteSpec.kt:15]: I also don't use star imports. Sorry for missing it.

**Comment by tibtof** [comment]
Hi @nomisRev, I implemented your suggestions, refactored the rest of the assertions and rebased the latest main. Please review it again. Thanks!

**Review by nomisRev — APPROVED** [review]
Looks great! Thank you @tibtof.
Sorry for the late response, I needed some time off.

**Pull request merged**